src/rdb.c

```diff
@@ -1445,6 +1445,7 @@ int rdbSave(int req, char *filename, rdbSaveInfo *rsi) {
         stopSaving(0);
         return C_ERR;
     }
+    if (fsyncFileDir(filename) == -1) goto werr;
```

Comment:
maybe each of these `goto werr` should set some string variable, so that the error log contains an indication of which call failed? (one of two fsync, fclose, etc)